### PR TITLE
OpenFoamHandler

### PR DESCRIPTION
**Pull request by fAndreuzzi**
In this commit I added the module `openfoamhandler` and some tests.

### CODE REVIEW

File: smithers/io/openfoamhandler.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+from operator import itemgetter
+
+
+class OpenFoamHandler:
+    """
+    Handler for OpenFOAM output files, based on the project Ofpp.
+    """
+
+    @classmethod
+    def _build_boundary(cls, mesh, key):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:14]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+        # extract faces which compose the boundary
+        bd_faces = np.array(mesh.faces[data.start : data.start + data.num])
+        # extract a list of unique points which compose the boundary
+        bd_points = np.unique(np.concatenate(bd_faces))
+
+        return {"faces": bd_faces, "points": bd_points, "type": data.type}
+
+    @classmethod
+    def _build_cells(cls, mesh, cell_idx):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:25]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+        return {
+            "faces_indexes": mesh.cell_faces[cell_idx],
+            "faces": cell_faces,
+            "points": cell_points,
+            "neighbours": mesh.cell_neighbour[cell_idx],
+        }
+
+    @classmethod
+    def _find_time_instants_subfolders(cls, path, fields_time_instants):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:39]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        elif isinstance(fields_time_instants, list):
+            return map(full_path_with_label, fields_time_instants)
+        else:
+            raise ValueError(
+                """Invalid value for the argument
+                `fields_time_instants`"""
+            )
+
+    @classmethod
+    def _find_fields_files(cls, fields_root_path, field_names):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:80]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+            return map(
+                full_path_with_name, next(os.walk(fields_root_path))[2]
+            )
+        elif isinstance(field_names, list):
+            return map(full_path_with_name, field_names)
+        else:
+            raise ValueError("Invalid value for the argument `field_names`")
+
+    @classmethod
+    def _no_fail_boundary_field(cls, path):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:96]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+    @classmethod
+    def _no_fail_internal_field(cls, path):
+        try:
+            return Ofpp.parse_internal_field(path)
+        except IndexError:
+            return None
+
+    @classmethod
+    def _load_fields(cls, time_instant_path, field_names):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:110]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+    @classmethod
+    def _no_fail_boundary_field(cls, path):
+        try:
+            return Ofpp.parse_boundary_field(path)
+        except IndexError:
+            return None
+
+    @classmethod
+    def _no_fail_internal_field(cls, path):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:103]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+        return dict(
+            (name,
+                (cls._no_fail_boundary_field(field_path),
+                    cls._no_fail_internal_field(field_path)))
+            for name, field_path in field_files
+        )
+
+    @classmethod
+    def _build_time_instant_snapshot(cls, ofpp_mesh, time_instant_path,
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:121]: Please, add a small docstring with types specification

File: smithers/io/openfoamhandler.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            },
+            "cells": {
+                cell_id: cls._build_cells(ofpp_mesh, cell_id)
+                for cell_id in range(len(ofpp_mesh.cell_faces))
+            },
+            "fields": cls._load_fields(time_instant_path, field_names)
+        }
+
+    @classmethod
+    def read(cls, filename, fields_time_instants="first", field_names="all"):
```
Comment (ndem0) [inline smithers/io/openfoamhandler.py:139]: Please, add a small docstring with types specification

**Comment by fAndreuzzi** [comment]
In 57d7cbb I fixed the issues you mentioned, if you think docstrings are OK I can squas my changes.

**Pull request merged**